crucible-syntax/src/Lang/Crucible/Syntax/Prog.hs

```diff
@@ -91,19 +92,24 @@ doParseCheck fn theInput pprint outh =
                       hPutStrLn outh $ show $ C.ppCFG' True (postdomInfo ssa) ssa
 
 -- | Allows users to hook into the various stages of 'simulateProgram'.
-data SimulateProgramHooks = SimulateProgramHooks
-  { setupOverridesHook ::
-      forall p sym ext t st fs. (IsSymInterface sym, sym ~ (ExprBuilder t st fs)) =>
+data SimulateProgramHooks ext = SimulateProgramHooks
+  { setupHook ::
+      forall p sym rtp a r t st fs. (IsSymInterface sym, sym ~ ExprBuilder t st fs) =>
+        sym ->
+        HandleAllocator ->
+        OverrideSim p sym ext rtp a r ()
```

Comment:
Add a comment here saying what `setupHook`'s purpose is. It might also be worth citing the specific example of `crucible-llvm-syntax`'s `setupHook` as an example of how it's used in practice.